include/zephyr/storage/stream_flash.h

```diff
@@ -106,9 +106,11 @@ size_t stream_flash_bytes_written(struct stream_flash_ctx *ctx);
  * @param data data to write
  * @param len Number of bytes to write
  * @param flush when true this forces any buffered data to be written to flash
- *        A flush write should be the last write operation in a sequence of
- *        write operations for given context (although this is not mandatory
- *        if the total data size is a multiple of the buffer size).
+ *        A write with the @p flush set to true has to be issued as the last
+ *        write request for a given context, as it concludes write of a stream;
+ *        there must not be issued any more write requests for given context,
+ *        unless it is re-initialized, and such write attempts may result in the
+ *        function returning error.
```

Comment:
this seems redundant with L102-103. I would completely remove this section and instead only update L102-103 (also maybe make it a `@warning` to make it stand out more?) to add the extra explanation regarding the expected behavior after a flush write. Maybe

Reply:
@kartben Does this look better now?